scripts/apps/authoring/attachments/reducer.ts

```diff
@@ -0,0 +1,38 @@
+import {
+    INIT_ATTACHMENTS,
+    ADD_ATTACHMENTS,
+    REMOVE_ATTACHMENT,
+    EDIT_ATTACHMENT,
+    SAVE_ATTACHMENT,
+} from './actions';
+
+const initialState = {
+    edit: null,
+    files: [],
+    maxSize: 0,
+    maxFiles: 10,
+};
+
+export function attachments(state = initialState, action: {type: string, payload: any}) {
+    switch (action.type) {
+    case INIT_ATTACHMENTS:
+        return {...action.payload};
+    case ADD_ATTACHMENTS:
+        return {...state, files: state.files.concat(action.payload)};
+    case REMOVE_ATTACHMENT:
+        return {...state, files: state.files.filter((f) => f !== action.payload)};
+    case EDIT_ATTACHMENT:
+        return {
+            ...state,
+            edit: action.payload,
```

Comment:
I would like naming to be improved:
EDIT_ATTACHMENT -> ENABLE_EDIT_MODE_FOR_ATTACHEMENT
SAVE_ATTACHMENT -> UPDATE_ATTACHEMENT